region/brinstar/pink.json

```diff
@@ -72,12 +72,28 @@
             {
               "id": 3,
               "requires": [
-                {"canShineCharge": {
-                  "usedTiles": 31
-                }}
+                {"and": [
+                  {"canShineCharge": {
+                    "usedTiles": 31
+                  }},
+                  {"or": [
+                    {"cost": [
+                      {"shinesparkFrames": 115}
+                    ]},
+                    {"and": [
+                      {"or": [
+                        "HiJump",
+                        "canWalljump"
+                      ]},
+                      {"cost": [
+                        {"shinesparkFrames": 95}
+                      ]}
+                    ]}
+                  ]}
+                ]}
               ],
               "unlock": null,
-              "note": "It's possible to get this shinespark doable with just no e-tanks with crazy good walljumping, or with HiJump and just good walljumping"
+              "note": "It's possible to get this shinespark doable with just no e-tanks with good walljumping, or with HiJump and walljumping"
```

Comment:
Shinesparks cut off at 29 health so getting out without e-tanks requires making it high enough to cut the shinespark frames to 70. With that in mind, you should take another look at the whole commit.

Reply:
True, forgot about that part. Will investigate.

Reply:
If you have an energy tank it won't cut out at 70 but use up the amount needed to the ceiling. This unregistered health drain while small might still be needed in future rooms causing a potentially broken seed so I will not include the InsaneWalljump but keep a similar note about it.

Reply:
Do you know how many frames there are between getting out and hitting the ceiling? I think that energy difference can likely be covered by health drops that the logic won't consider.